utils/dataset_utils.py

```diff
@@ -1,20 +1,134 @@
 import tensorflow as tf
+import tensorflow.contrib as tf_contrib
 import numpy as np
+from tensor2tensor.data_generators.speech_recognition import SpeechRecognitionProblem
+from tensor2tensor.models.transformer import transformer_librispeech_tpu
+import utils
 
 __all__ = [
-    'read_dataset',
-    'process_dataset',
+    'input_fn',
+    'input_fn_t2t'
 ]
 
+def read_dataset_t2t_format(data_dir, num_parallel_calls, mode, max_frames, max_symbols):
+    problem = SpeechRecognitionProblem()
+    problem.name = 'asr_ipa_precalc'
```

Comment:
Probably we should support other problems, e.g. librispeech. So it might be reasonable to move problem name to configuration.

Reply:
So far we don't support `librispeech`, because its dataset format is different from ours. Targets in `librispeech` are decoded as UTF-8 bytes, without using any vocabulary. If we  decide to add support to `librispeech` one day, we'll need to do much more changes to the code than just moving problem name to config. BTW, not only `asr_ipa_precalc` problem is supported now, but all problems which use `IPAEncoder`.

Reply:
UTF-8 bytes is not a problem. We could just use 256 vocabulary size.
Hardcoding for `asr_ipa_precalc` problem makes things rather complicated, because this way this (public!) repository depends not only on tensor2tensor, but also on t2t_problems. Datapreparation becomes a very convoluted process.

Reply:
`problem.name` is only used for fie names :)

Reply:
Then it definitely should be a configurable option!